src/index.js

```diff
@@ -59,69 +59,97 @@ export function asArray(cleaner) {
       throw new TypeError('Expected an array')
     }
 
+    var out = []
     try {
-      var out = []
       for (var i = 0; i < raw.length; ++i) {
         out[i] = cleaner(raw[i])
       }
-      return out
     } catch (error) {
       throw locateError(error, '[' + i + ']', 0)
     }
+    return out
   }
 }
 
 /**
- * Makes a cleaner that accepts an object with arbitrary keys
- * and the given value type. Removes keys named `__proto__` for safety.
+ * Makes a cleaner that accepts an object.
+ *
+ * If asObject receives a single cleaner function,
+ * it will will use that to clean all the object's own enumerable properties
+ * (except "__proto__", which gets filtered out).
+ *
+ * Otherwwise, if asObject receives a an object with cleaners as properties,
+ * it will apply each cleaner to the matching key in the object.
  */
-export function asMap(cleaner) {
-  return function asMap(raw) {
-    if (raw == null || typeof raw !== 'object') {
-      throw new TypeError('Expected an object')
-    }
+export function asObject(shape) {
+  // The key-value version:
+  if (typeof shape === 'function') {
+    return function asObject(raw) {
+      if (typeof raw !== 'object' || raw == null) {
+        throw new TypeError('Expected an object')
+      }
 
-    var keys = Object.keys(raw)
-    try {
       var out = {}
-      for (var i = 0; i < keys.length; ++i) {
-        var key = keys[i]
-        if (key === '__proto__') continue
-        out[key] = cleaner(raw[key])
+      var keys = Object.keys(raw)
+      try {
+        for (var i = 0; i < keys.length; ++i) {
+          var key = keys[i]
+          if (key === '__proto__') continue
```

Comment:
When does `Object.keys` include `'__proto__'` in it's output? After checking this, it looks like Object.keys will not include properties `p` where `.hasOwnProperty(p)` returns 'false'.

Reply:
You can get this from `JSON.parse('{ "__proto__": 1 }')`. It's actually a huge source of security vulnerabilities in the ecosystem, so we eliminate that as part of cleaning.